src/Simplex/Messaging/Agent.hs

```diff
@@ -312,9 +351,26 @@ acceptInv c connId invId connInfo =
 subscribeConnection' :: AgentMonad m => AgentClient -> ConnId -> m ()
 subscribeConnection' c connId =
   withStore (`getConn` connId) >>= \case
-    SomeConn _ (DuplexConnection _ rq _) -> subscribeQueue c rq connId
+    SomeConn _ (DuplexConnection _ rq sq) -> case status (sq :: SndQueue) of
+      Confirmed -> do
+        let verifyKey = C.publicKey $ sndPrivateKey sq
+        case verifyKey of
+          Nothing -> throwError $ CONN NO_PUBLIC_KEY
```

Comment:
I would simply return INTERNAL here, it indicates a logical implementation error, not a protocol error

Reply:
also line 528 - Confirmation -> ApprovedConfirmation w/t Maybe

Reply:
done